lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -169,19 +138,26 @@ impl GelfDeserializer {
                 if key == "_id" {
                     continue;
                 }
+                // per GELF spec, Additional field names must be prefixed with an underscore
+                if !key.starts_with('_') {
+                    return Err(format!(
+                        "'{}' field is invalid. \
+                                       Additional field names must be prefixed with an underscore.",
+                        key
+                    )
+                    .into());
+                }
                 // per GELF spec, Additional field names must be characters dashes or dots
                 if !VALID_FIELD_REGEX.is_match(key) {
-                    if key.starts_with('_') {
-                        return Err(format!("'{}' field is invalid. \
-                                           Additional field names must be prefixed with an underscore.", key).into());
-                    } else {
-                        return Err(format!("'{}' field contains invalid characters. Field names may \
-                                           contain only letters, numbers, underscores, dashes and dots.", key).into());
-                    }
+                    return Err(format!("'{}' field contains invalid characters. Field names may \
+                                       contain only letters, numbers, underscores, dashes and dots.", key).into());
                 }
 
-                let vector_val: value::Value = val.into();
-                log.insert(path!(key.as_str()), vector_val);
+                // per GELF spec, Additional field values must be either strings or numbers
+                if val.is_string() || val.is_number() {
+                    let vector_val: value::Value = val.into();
+                    log.insert(path!(key.as_str()), vector_val);
+                }
```

Comment:
In other places we are outright rejecting if a field is invalid (e.g. the field name). For consistency, I think we should do that here too.

Reply:
This actually made me realize we can leverage the `internal_events` that I added in the encoder PR here  ... but the encoder PR is setup to depend on this PR. doh!

Reply:
Actually, I think you just want to return an error here. I'm 95% sure (but we should verify) that the codec handling framework emits the internal events if a codec returns an error during decoding.

Reply:
Yeah, it happens here: https://github.com/vectordotdev/vector/blob/24b56679865ab3cb63965292ce88462a6305ab43/src/codecs/decoding/decoder.rs#L66

Reply:
Ooo shoot. So the changes I have made in the encoder probably need to be scrapped I think.

Reply:
Ok, I cleaned up the encoder PR to remove internal_event emission.
And in the decoder, I added an Err for if the value type is incorrect.